src/content/getusermedia/resolution/js/main.js

```diff
@@ -60,10 +77,26 @@ function displayVideoDimensions() {
   }
   dimensions.innerHTML = 'Actual video dimensions: ' + video.videoWidth +
     'x' + video.videoHeight + 'px.';
+  widthInput.value = video.videoWidth;
+  widthOutput.textContent = video.videoWidth;
 }
 
 video.onloadedmetadata = displayVideoDimensions;
 
+function constraintChange(e) {
+  widthOutput.textContent = e.target.value;
+  let track = window.stream.getVideoTracks()[0];
+  let constraints = {video: {width: {exact: e.target.value}}};
```

Comment:
This shouldn't have  the "video:" attribute since the track already knows that it's a video track.
|constraints| should be {width: {exact: e.target.value}}

Reply:
Note Firefox only supports native resolutions, so it will reject if there is no native mode with the given target value.